Apply CRUD to Webhooks admin

claudiosanches opened the pull request:
Closes #17587

claudiosanches committed:
Apply Webhook CRUD to WC_Admin_Webhooks

claudiosanches committed:
Fixed coding standards in includes/admin/settings/class-wc-settings-api.php

claudiosanches committed:
Always 'post' on API since there is no more Trash for webhooks

claudiosanches committed:
Apply Webhooks CRUD and fixed coding standards in html-webhooks-edit.php

claudiosanches committed:
Merge branch 'feature/webhook-crud' into feature/12439-2

claudiosanches committed:
Fixed coding standards in webhooks table list class

claudiosanches committed:
Added function to get webhook instance

claudiosanches committed:
Search webhooks with data store

claudiosanches committed:
Fixed coding standards

claudiosanches committed:
Cache search webhooks results and added method to count webhooks by status

claudiosanches committed:
Fixed coding standards in WC_Admin_Menus class

claudiosanches committed:
Introduced new filters to allow save settings

claudiosanches committed:
Fixed coding standards for WC_Admin_Settings

claudiosanches committed:
New settings REST API file

claudiosanches committed:
Fixed WC_Webhook_Data_Store::get_webhooks_ids transient

claudiosanches committed:
Properly allow to POST data in table list without trigger save settings

claudiosanches committed:
Implements Webhook CRUD into table list

claudiosanches committed:
Removed "trash" actions and included new "delete" actions

mikejolley commented:
@claudiosanches This looks good to go, but there are 3 things bothering me in admin for webhooks which I feel need looking at.

1. Creating/adding a webhook creates it right away and does not require save. Posts for example don't do this - even if a draft is created in the DB it's not visible.
2. The dates in the "Webhook actions" section - why are there here? If they need to be here, there is a bug with updated. On a new webhook before save it shows a date in 2001. If they don't need to be here, move to list table.
3. Webhook logs shows nothing when empty.

claudiosanches committed:
Stop creating webhooks when hit the "Add webhook" button

Creates only after save edited webhook.

claudiosanches committed:
Fixed how read empty dates while reading webhooks from database

claudiosanches commented:
@mikejolley 

> 1. Creating/adding a webhook creates it right away and does not require save. Posts for example don't do this - even if a draft is created in the DB it's not visible.

Done.

> 2. The dates in the "Webhook actions" section - why are there here? If they need to be here, there is a bug with updated. On a new webhook before save it shows a date in 2001. If they don't need to be here, move to list table.

Need to display this date in someplace.
Fixed already the bug with dates and I can include in the table list too.

> 3. Webhook logs shows nothing when empty.

There is no more logs now, since will be saved using the WooCommerce log system, so maybe we should link into the files there.
But I just removed the HTML now.

claudiosanches committed:
Fixed how check for webhooks dates

claudiosanches committed:
Fixed how update at date displays
